tests/benchdnn/graph/input_displacer.cpp

```diff
@@ -615,4 +642,63 @@ int partition_data_displacer_t::gen_causal_mask_filling(
     return OK;
 }
 
+int partition_data_displacer_t::gen_softmax_stats_filling(
+        const ::graph::deserialized_op_t &main_op, int arg,
+        const dnn_mem_t &src_mem, dnn_mem_t &mem, const_dnnl_memory_desc_t md,
+        res_t *res) const {
+
+    dnn_mem_t m(md, get_test_engine(), /* prefill = */ false);
+    std::vector<float> dst(src_mem.nelems(), 0.f);
+
+    logical_tensor::dims softmax_src_shape = main_op.in_lts_[0].shape_;
+    logical_tensor::dims softmax_stats_shape = main_op.in_lts_[1].shape_;
+    size_t axis = 0;
+    for (; axis < softmax_src_shape.size() && axis < softmax_src_shape.size();
+            ++axis) {
+        if (softmax_src_shape[axis] != softmax_stats_shape[axis]) break;
+    }
+
+    int64_t outer_size {1}, inner_size {1}, axis_size {1};
+    for (size_t i = 0; i < axis; i++) {
+        outer_size *= softmax_src_shape[i];
+    }
+    for (size_t i = axis + 1; i < softmax_src_shape.size(); i++)
+        inner_size *= softmax_src_shape[i];
+    axis_size = softmax_src_shape[axis];
+
+    benchdnn_parallel_nd(outer_size, inner_size, [&](int64_t ou, int64_t in) {
+        float space_denom = 0.;
+        float space_max = -FLT_MAX;
+        int64_t ou_in_offset = ou * axis_size * inner_size + in;
+
+        for (int64_t as = 0; as < axis_size; ++as) {
+            int64_t idx = ou_in_offset + as * inner_size;
+            space_max = MAX2(space_max, src_mem.get_f32_elem(idx));
+        }
+
+        for (int64_t as = 0; as < axis_size; ++as) {
+            int64_t idx = ou_in_offset + as * inner_size;
+            float s = src_mem.get_f32_elem(idx);
+            float D = dst[idx] = expf(s - space_max);
+            space_denom += D;
+        }
+
+        space_denom = space_denom ? 1.f / space_denom : 1.f;
+        float dst_space_max = -FLT_MAX;
+        for (int64_t as = 0; as < axis_size; ++as) {
+            int64_t idx = ou_in_offset + as * inner_size;
+            dst[idx] *= space_denom;
+            dst_space_max = MAX2(dst_space_max, dst[idx]);
+        }
+
+        // computes stats
+        int64_t stats_idx = ou * inner_size + in;
+        float stats_value = dst_space_max ? space_max - logf(space_denom) : 0.f;
+        m.set_elem(stats_idx, stats_value);
+    });
```

Comment:
Could you clarify, please, why exactly these values are needed? It seems to me that it's only a backward path that gets executed here, thus, data from forward is not available and final values don't have to have anything in common with forward, unless I'm missing something...